migrate/src/main/scala/migrate/MigrateLib.scala

```diff
@@ -0,0 +1,142 @@
+import migrate.utils.CoursierHelper
+
+sealed trait MigrateLib extends Lib {
+  val organization: Organization
+  val name: Name
+  val revision: Revision
+  val crossVersion: CrossVersion
+
+  override def getOrganization: String = organization.value
+  override def getName: String         = name.value
+  override def getRevision: String     = revision.value
+  override def getCrossVersion: String = crossVersion.toString
+
+  override def toString: String = s"${organization.value}:${name.value}:${revision.value}"
+
+}
+
+case class Lib213(organization: Organization, name: Name, revision: Revision, crossVersion: CrossVersion)
+    extends MigrateLib {
+  def toCompatible: Seq[CompatibleWithScala3Lib] =
+    crossVersion match {
+      // keep the same if CrossVersion.Disabled. Usually it's a Java Lib
+      case CrossVersion.Disabled => Seq(CompatibleWithScala3Lib.from(this))
+      // look for revisions that are compatible with scala 3 binary version
+      case CrossVersion.Binary(_, _) => getCompatibleWhenBinaryCrossVersion()
+      // look for revisions that are compatible with scala 3 full version
+      case CrossVersion.Full(_, _) => CoursierHelper.getCompatibleForScala3Full(this)
+      // already compatible
+      case CrossVersion.For2_13Use3(_, _) => Seq(CompatibleWithScala3Lib.from(this))
+      case CrossVersion.For3Use2_13(_, _) => Seq(CompatibleWithScala3Lib.from(this))
+      // For Patch and Constant, we search full compatible scala 3 version
+      case CrossVersion.Patch       => CoursierHelper.getCompatibleForScala3Full(this)
+      case CrossVersion.Constant(_) => CoursierHelper.getCompatibleForScala3Full(this)
+    }
+
+  private def getCompatibleWhenBinaryCrossVersion(): Seq[CompatibleWithScala3Lib] = {
+    val scala3Libs = CoursierHelper.getCompatibleForScala3Binary(this)
+    if (scala3Libs.isEmpty) {
+      if (macroLibs.get(this.organization).contains(this.name)) Nil
+      else CoursierHelper.getCompatibleForBinary213(this)
+    } else scala3Libs
+  }
+
+}
+
+case class CompatibleWithScala3Lib(
+  organization: Organization,
+  name: Name,
+  revision: Revision,
+  crossVersion: CrossVersion
+) extends MigrateLib {}
+
+object MigrateLib {
+  case class Organization(value: String)
+  case class Name(value: String)
+  case class Revision(value: String) extends Ordered[Revision] {
+    override def compare(that: Revision): Int =
+      Revision.ordering.compare(this, that)
+    private val version: Seq[String] = value.split('.')
+    val major: Option[Int]           = version.headOption.flatMap(v => Try(v.toInt).toOption)
+    val minor: Option[Int]           = Try(version(1).toInt).toOption
+    val patch: Option[Int]           = Try(version(2).split("-")(0).toInt).toOption
+    val beta: Option[String]         = Try(version(2).split("-")(1)).toOption
+
+  }
+  object Revision {
+    implicit val ordering: Ordering[Revision] = Ordering.by { x: Revision =>
+      (x.major, x.minor, x.patch, x.beta)
```

Comment:
This is a `Revision => (Option[Int], Option[Int], Option[Int],  Option[String])`, how is it resolved to a proper ordering?

Reply:
I have to test it ^^